src/engine/strat_engine/metadata.rs

```diff
@@ -4,262 +4,365 @@
 
 use std::cmp::Ordering;
 use std::str::from_utf8;
+use std::fs::File;
+use std::io::{Read, Seek, SeekFrom, Write};
 
 use byteorder::ByteOrder;
 use byteorder::LittleEndian;
-
 use crc::crc32;
-
 use time::Timespec;
-
 use uuid::Uuid;
 
-use consts::SECTOR_SIZE;
-
 use types::Sectors;
-use types::SectorOffset;
-
 use super::engine::DevOwnership;
+use consts::SECTOR_SIZE;
+use engine::{EngineResult, EngineError, ErrorEnum};
+use super::blockdev::BDA_STATIC_HDR_SIZE;
```

Comment:
Don't know why BDA_STATIC_HDR_SIZE isn't defined in metadata and used in blockdev.